shared_model/interfaces/iroha_internal/proposal.hpp

```diff
@@ -47,16 +48,28 @@ namespace shared_model {
             and createdTime() == rhs.createdTime();
       }
 
+      virtual const types::BlobType &blob() const = 0;
+
+      const types::HashType &hash() const {
```

Comment:
Looks like `class Signable` need to be splitted to Signable & Hashable. What do you think about it?

Reply:
I guess not, because they were merged in the past https://github.com/hyperledger/iroha/pull/1131

Reply:
:( and now they're needed